crates/dojo/core-cairo-test/src/tests/helpers.cairo

```diff
@@ -262,3 +266,182 @@ pub fn drop_all_events(address: ContractAddress) {
         };
     }
 }
+
+/// Note: These models must remain private to be able to redefine them
+/// in `resources.cairo` test file, to test model upgrade.
+#[derive(Copy, Drop, Serde, Debug)]
+#[dojo::model]
+struct FooNameChanged {
+    #[key]
+    pub caller: ContractAddress,
+    pub a: felt252,
+    pub b: u128,
+}
+
+#[derive(Copy, Drop, Serde, Debug)]
+#[dojo::model]
+struct FooBadLayoutType {
+    #[key]
+    pub caller: ContractAddress,
+    pub a: felt252,
+    pub b: u128,
+}
+
+#[derive(Copy, Drop, Serde, Debug)]
+#[dojo::model]
+struct FooMemberRemoved {
+    #[key]
+    pub caller: ContractAddress,
+    pub a: felt252,
+    pub b: u128,
+}
+
+#[derive(Copy, Drop, Serde, Debug)]
+#[dojo::model]
+struct FooMemberAddedButRemoved {
+    #[key]
+    pub caller: ContractAddress,
+    pub a: felt252,
+    pub b: u128
+}
+
+#[derive(Copy, Drop, Serde, Debug)]
+#[dojo::model]
+struct FooMemberAddedButMoved {
+    #[key]
+    pub caller: ContractAddress,
+    pub a: felt252,
+    pub b: u128,
+}
+
+#[derive(Copy, Drop, Serde, Debug)]
+#[dojo::model]
+struct FooMemberAddedButSameVersion {
+    #[key]
+    pub caller: ContractAddress,
+    pub a: felt252,
+    pub b: u128
+}
+
+#[derive(Copy, Drop, Serde, Debug)]
+#[dojo::model]
+struct FooMemberAdded {
+    #[key]
+    pub caller: ContractAddress,
+    pub a: felt252,
+    pub b: u128
+}
+
+pub fn deploy_world_for_model_upgrades() -> IWorldDispatcher {
+    let namespace_def = NamespaceDef {
+        namespace: "dojo", resources: [
+            TestResource::Model(foo::TEST_CLASS_HASH.try_into().unwrap()),
+            TestResource::Model(foo_name_changed::TEST_CLASS_HASH.try_into().unwrap()),
+            TestResource::Model(foo_bad_layout_type::TEST_CLASS_HASH.try_into().unwrap()),
+            TestResource::Model(foo_member_removed::TEST_CLASS_HASH.try_into().unwrap()),
+            TestResource::Model(foo_member_added_but_moved::TEST_CLASS_HASH.try_into().unwrap()),
+            TestResource::Model(foo_member_added_but_removed::TEST_CLASS_HASH.try_into().unwrap()),
+            TestResource::Model(
+                foo_member_added_but_same_version::TEST_CLASS_HASH.try_into().unwrap()
+            ),
+            TestResource::Model(foo_member_added::TEST_CLASS_HASH.try_into().unwrap())
+        ].span()
+    };
+    spawn_test_world([namespace_def].span())
+}
+
+/// Note: These events must remain private to be able to redefine them
+/// in `resources.cairo` test file, to test event upgrade.
```

Comment:
Could you detail on why they have to be redefined and not used from here directly?